src/contracts/pods/EigenPodManager.sol

```diff
@@ -243,6 +541,120 @@ contract EigenPodManager is Initializable, OwnableUpgradeable, Pausable, IEigenP
+            IStrategy[] memory strategies = new IStrategy[](1);
+            strategies[0] = beaconChainETHStrategy;
+            delegationManager.decreaseDelegatedShares(podOwner, strategies, shareAmounts);
+        }
+    }
+
+    // @notice Changes the `podOwner`'s shares by `sharesDelta` and performs a call to the DelegationManager to ensure delegated shares are also tracked correctly
+    function _recordBeaconChainETHBalanceUpdate(address podOwner, int256 sharesDelta) internal {
+        if (sharesDelta < 0) {
+            // if change in shares is negative, remove the shares
+            _removeShares(podOwner, uint256(-sharesDelta));
+        } else {
+            // if change in shares is positive, add the shares
+            _addShares(podOwner, uint256(sharesDelta));
+        }      
+    }
+
+    /**
+     * @notice This function is called to decrement withdrawableRestakedExecutionLayerGwei when a validator queues a withdrawal.
+     * @param amountWei is the amount of ETH in wei to decrement withdrawableRestakedExecutionLayerGwei by
+     */
+    function _decrementWithdrawableRestakedExecutionLayerGwei(address podOwner, uint256 amountWei) 
+        internal
+    {
+        ownerToPod[podOwner].decrementWithdrawableRestakedExecutionLayerGwei(amountWei);
+    }
+
+    /**
+     * @notice Withdraws ETH from an EigenPod. The ETH must have first been withdrawn from the beacon chain.
+     * @param podOwner The owner of the pod whose balance must be withdrawn.
+     * @param recipient The recipient of the withdrawn ETH.
+     * @param amount The amount of ETH to withdraw.
+     * @dev Callable only by the StrategyManager contract.
+     */
+    function _withdrawRestakedBeaconChainETH(address podOwner, address recipient, uint256 amount)
+        internal onlyWhenNotPaused(PAUSED_WITHDRAW_RESTAKED_ETH)
+    {
+        ownerToPod[podOwner].withdrawRestakedBeaconChainETH(recipient, amount);
+    }
+
+    /**
+     * @notice If the `staker` has no existing shares, then they can `undelegate` themselves.
+     * This allows people a "hard reset" in their relationship with EigenLayer after withdrawing all of their stake.
+     * @param staker The address to undelegate. Passed on as an input to the `delegationManager.undelegate` function.
+     */
+    function _undelegate(address staker) internal onlyNotFrozen(staker) {
+        require(stakerCanUndelegate(staker), "EigenPodManager._undelegate: staker has active deposits");
+        delegationManager.undelegate(staker);
+    }
+
+    /**
+     * @notice Internal function to enter `podOwner` into undelegation limbo
+     * @dev Does nothing if the `podOwner` has no delegated shares (i.e. they are already in undelegation limbo or have no shares)
+     * OR if they are not actively delegated to any operator.
+     */
+    function _enterUndelegationLimbo(address podOwner) internal {
+        if (!stakerHasNoDelegatedShares(podOwner) && delegationManager.isDelegated(podOwner)) {
```

Comment:
not using a require statement here means the `forceIntoUndelegationLimbo` function will not revert in weird edge cases (where it might, with the require). Open to suggestions, but I specifically made this change on purpose, in this commit https://github.com/Layr-Labs/eigenlayer-contracts/pull/135/commits/f08e02560d726d7abea29685da8671150784e3a8